recipes/dd-opentracing-cpp/all/conanfile.py

```diff
@@ -22,99 +29,80 @@ class DatadogOpenTracingConan(ConanFile):
     }
 
-    generators = "cmake", "cmake_find_package"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
     @property
     def _compilers_minimum_version(self):
         return {
             "gcc": "5",
             "Visual Studio": "15",
+            "msvc": "191",
             "clang": "3.4",
             "apple-clang": "7",
         }
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("opentracing-cpp/1.6.0")
-        self.requires("zlib/1.2.11")
-        self.requires("libcurl/7.80.0")
-        self.requires("msgpack/3.3.0")
-        self.requires("nlohmann_json/3.10.5")
+        self.requires("opentracing-cpp/1.6.0", transitive_headers=True, transitive_libs=True)
+        self.requires("zlib/[>=1.2.11 <2]")
+        self.requires("libcurl/[>=7.78 <9]")
+        self.requires("msgpack-cxx/6.1.0")
+        self.requires("nlohmann_json/3.11.2")
 
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 14)
+            check_min_cppstd(self, 14)
 
         minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
         if minimum_version:
-            if tools.Version(self.settings.compiler.version) < minimum_version:
-                raise ConanInvalidConfiguration("Datadog-opentracing requires C++14, which your compiler does not support.")
+            if Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    "Datadog-opentracing requires C++14, which your compiler does not support."
+                )
         else:
-            self.output.warn("Datadog-opentracing requires C++14. Your compiler is unknown. Assuming it supports C++14.")
+            self.output.warning(
+                "Datadog-opentracing requires C++14. Your compiler is unknown. Assuming it supports C++14."
+            )
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if self.settings.compiler == "clang" and Version(self.settings.compiler.version) < "4.0":
```

Comment:
Could you please add a commend like `# INFO: ` why is it needed now?

Reply:
I don't remember the exact reason anymore. I tested the older versions in a Docker container and that was the latest supported version.

Reply:
It still miss information, I see no relation in the upstream. Please, remove it.